## Intervalo de confianza de la media con varianza conocida

La empresa AUTOZIN produce partes de componentes cilíndricos para la industria automotriz que tienen un promedio de diámetro de 5 milímetros con una distribución normal y una desviación estándar de 0.1 milímetros. Para confirmar esto se tomó una muestra de 25 partes seleccionadas al azar que muestra que el diámetro promedio es de 5.027 milímetros. Encuentre el intervalo de confianza del 95% para la media de la población de todas las partes de componentes cilíndricos que produce la empresa.

**Se empleara:**
$$
IC{:}\quad\bar{X}-Z_{(1-\frac\alpha2)}\cdot\frac\sigma{\sqrt{n}}\leq\mu\leq\bar{X}+Z_{(1-\frac\alpha2)}\cdot\frac\sigma{\sqrt{n}}
$$

Los parámetros relevantes son:
- Varianza Poblacional ( $\sigma$ ) = 0.1
- Promedio ($\bar{X}$) = 5.027
- Tamaño de muestra ($n$) = 25
- Confianza = 95 %

**Si:**
1 - $\alpha$ = 0.95

**Entonces:**
- $\alpha$ = 0.05

In [4]:
import scipy.stats as st
import math
# Asignando valores:
O = 0.1
X = 5.027
n = 25
# a -> calculado previamente
a = 0.05

# Calculamos Z
Z = round((st.norm.ppf(1-(a/2))),2)

# Reemplazamos en la formula IC
IC1 = X-(Z*O/math.sqrt(n)) 
IC2 = X+(Z*O/math.sqrt(n)) 

print(Z)
print("IC:",IC1," u ",IC2)

1.96
IC: 4.9878  u  5.0662


Con un nivel de confianza del 95%, el diámetro promedio de los componentes cilíndricos está comprendido desde 4.9878 a 5.0662 mil.

---

## Intervalo de confianza de la media con varianza desconocida
En una muestra aleatoria de 20 porciones de cereal, el contenido promedio de azúcar fue de 11.3 gramos con una desviación estándar de 2.45 gramos. Suponiendo que los contenidos de azúcar están distribuidos normalmente, determine el intervalo de confianza del 95% para el contenido promedio de azúcar en las porciones de dicho cereal.

**Se usara:**
$$
\mathsf{IC}{:}\quad\bar{X}-T_{(1-\frac{\alpha}{2},gl)}\cdot\frac{S}{\sqrt{n}}\leq\mu\leq\bar{X}+T_{(1-\frac{\alpha}{2},gl)}\cdot\frac{S}{\sqrt{n}}
$$

Los parámetros relevantes son:
- Desviacion estandar muestral ( $S$ ) = 2.45
- Promedio ($\bar{X}$) = 11.3
- Tamaño de muestra ($n$) = 20
- Confianza = 95 % 

**Si:**
1 - $\alpha$ = 0.95

**Entonces:**
- $\alpha$ = 0.05

In [1]:
import scipy.stats as st
import math

# Asignando valores:
S = 2.45
X = 11.3
n = 20
# a -> calculado previamente
a = 0.05
gl = n - 1

# Calculamos T
T = round(st.t.ppf(1 - a / 2, df=gl),3)

# Reemplazamos en la fórmula IC
IC1 = round(X - T * (S / math.sqrt(n)),3)
IC2 = round(X + T * (S / math.sqrt(n)),3)

print("T:", T)
print("Intervalo de confianza:", IC1, " a ", IC2)

T: 2.093
Intervalo de confianza: 10.153  a  12.447


Con una confianza del 95%, el contenido promedio de azúcar será de 10.153 a 12.447 gramos.

---

# Intervalos de confianza para la diferencia de medias muestrales con varianza conocida y desconocida.

**Conocida:**

En un estudio para determinar el gasto medio mensual de las ciudades A y B con desviaciones estandar de 15 y 10 soles respectivamente. Se toma una muestra al azar de 200 hogares de A arrojando un gasto medio de S/250. Una muestra al azar de 180 hogares de la ciudad B de un gasto medio de 235.

a) Determine en un intervalo de confianza del 99% para la diferencia de 
gasto medio en las ciudades A y B.

b) ¿Es diferente el gasto medio mensual en arbitrios en las ciudades A y B?

Los parametros relevantes son:
- Varianza Poblacional (Desviacion) ($\sigma_1^2$) = 15
- Muestra 1 ($n_1$) = 200 
- Promedio 1 ($\bar{X}_1$) = 250
- Varianza Poblacional (Desviacion) ($\sigma_2^2$) = 20
- Muestra 2 ($n_2$) = 180
- Promedio 2 ($\bar{X}_2$) = 235
- Confianza = 99%

**Formula a utilizar:**

$$
IC=(X_1-X_2)-Z_{(1-\frac{\alpha}{2})}\sqrt[]{\frac{\sigma_1{}^2}{n_1}+\frac{\sigma_2{}^2}{n_2}}\leq\boldsymbol{\mu_1}-\boldsymbol{\mu_2}\leq(X_1-X_2)+Z_{(1-\frac{\alpha}{2})}\sqrt[]{\frac{\sigma_1{}^2}{n_1}+\frac{\sigma_2{}^2}{n_2}}
$$

**Si:**
1 - $\alpha$ = 0.99

**Entonces:**
- $\alpha$ = 0.01

In [17]:
import scipy.stats as st
import math

# Asignando valores:
o1 = 15
o2 = 20
n1 = 200
n2 = 180
x1 = 250
x2 = 235
# a -> calculado previamente
a = 0.01

# Calculamos Z
Z = round(st.norm.ppf(1 - a / 2), 3)

# Reemplazamos en la fórmula IC
IC1 = round((x1 - x2) - Z * (math.sqrt(o1**2 / n1 + o2**2 / n2)), 3)
IC2 = round((x1 - x2) + Z * (math.sqrt(o1**2 / n1 + o2**2 / n2)), 3)

print("Z:", Z)
print("Intervalo de confianza:", IC1, "<= \u03BC <=", IC2)

Z: 2.576
Intervalo de confianza: 10.287 <= μ <= 19.713


Con un nivel de confianza del 99% podemos decir que la diferencia de medias de los gastos en arbitrios de las ciudades A y B se encuentran en el intervalo (10.287 y 19.713)

---

Desconocida:

En un proceso quimico, se comparan dos catalizadores para verificar su efecto en el resultado de la reaccion del proceso. Se preparo una muestra de 12 procesos utilizando el catalizador 1 y una de 10 con el catalizador 2, en el primer caso se obtuvo un rendimiento promedio de 85 con una desivacion estandar muestral de 4, mientras que en el promedio para la segunda muestra fue 81 y la desviacion estandar muestral de 5. Encuentre un intervalo de confianza del 90% para la diferencia entre las medias poblacionales, suponiendo que las **poblaciones** están dsitribuidas aproximadamente en **forma normal**, con **varianzas iguales**.

Los parametros relevantes son:
- Muestra 1 ($n_1$) = 12
- Promedio 1 ($\bar{X}_1$) = 85
- Varianza Muestral (${S_{1}}$): 4 
- Muestra 2 ($n_2$) = 10
- Promedio 2 ($\bar{X}_2$) = 81
- Varianza Muestral (${S_{2}}$): 5
- Confianza = 90%

**Formulas a usar:**

Se emplea la distribucion t-Student con $gl=n_1+n_2-2$

$$
{S_P}^2=\frac{(n_1-1){S_1}^2+(n_2-1){S_2}^2}{n_1+n_2-2}
$$

$$
(X_{1}-X_{2})-T_{(1-\frac{\alpha}{2},gl)}\sqrt{S_{P}^{2}(\frac{1}{n_{1}}+\frac{1}{n_{2}})}\le\mu_{1}-\mu_{2}\le(X_{1}-X_{2})+T_{(1-\frac{\alpha}{2},gl)}\sqrt{S_{P}^{2}(\frac{1}{n_{1}}+\frac{1}{n_{2}})}
$$

**Si:**
1 - $\alpha$ = 0.90

**Entonces:**
- $\alpha$ = 0.1

In [22]:
import scipy.stats as st
import math

# Asignando valores:
s1 = 4
s2 = 5
n1 = 12
n2 = 10
x1 = 85
x2 = 81
# a -> calculado previamente
a = 0.1
gl=n1+n2-2

# Calculamos T
T = round(st.t.ppf(1-(a/2),df=gl),4)

# Calculamos sp^2
Sp = ((n1 - 1) * s1**2 + (n2 - 1) * s2**2) / gl

# Aplicamos la Formula IC
IC1 = round((x1-x2)-T*(math.sqrt(Sp*(1/n1+1/n2))), 3)
IC2 = round((x1-x2)+T*(math.sqrt(Sp*(1/n1+1/n2))), 3)

print("T:", T)
print("SP:",Sp)
print("Intervalo de confianza:", IC1, "<= \u03BC <=", IC2)

T: 1.7247
SP 20.05
Intervalo de confianza: 0.693 <= μ <= 7.307


Con un nivel de confianza del 90% podemos concluir que la diferencia de medias del rendimiento de los catalizadores se encuentran en el intervalo (0.69 y 7.31)

---

# Intervalos de Confianza para la proporcion

Los compradores de ultima hora se quejan de que las tiendas por departamento de LIMA cierran muy temprano. En una muestra aleatoria de 600 compradores compulsivos de un centro comercial se encontro que 360 estan a favor de un horario más amplio para las compras.
Calcule e interprete en un intervalo del 95% de confianza para la proporcion verdadera de compradores que estan a favor de un horario más amplio para las compras.

Parametros resaltantes:
- Muestra ($n$) = 600
- Numero de exitos en la muestra ($X$) = 360
- Confianza = 95%

**Formulas a utilizar:**
$$
IC=p-Z_{\left(1-\frac\alpha2\right)^\bullet}\sqrt{\frac{pq}n}<{\pi}<p+Z_{\left(1-\frac\alpha2\right)^\bullet}\sqrt{\frac{pq}n}
$$

A favor:
$$
p=\frac{X}{n}
$$
En contra:
$$
q = 1 - p 
$$

In [24]:
import scipy.stats as st
import math

# Asignando valores:
n=600
X=360
CF=0.95
p=X/n
q=1-p
a = 1 - CF

# Calculamos Z
Z = round(st.norm.ppf(1-(a/2)),4)

# Aplicamos la formula IC
IC1 = round(p-Z*(math.sqrt(p*q/n)),3)
IC2 = round(p+Z*(math.sqrt(p*q/n)),3)

print("Z:", Z)
print("Intervalo de confianza:", IC1, "<= pi <=", IC2)

Z: 1.96
Intervalo de confianza: 0.561 <= pi <= 0.639


Con un nivel de confianza del 95% podemos concluir ue la verdadera prorcion de personas a favor de un horario mas amplio para compras se encuentra en el intervalo de (0.561,0.639)

---

# Intervalos de confianza para la diferencia de proporciones

En un proyecto de Ing. Industrial se ha lanzado una bebida energizante. En una zona muestra de 500 alumnos del Campus de Lima Sur se encontró que a 300 les agrada la nueva la nueva bebida y en una muestra de 400 de alumnos del Campus de Lima Norte se encontró que a 230 les agrada la nueva bebida. Calcule e interprete un intervalo del 95% de confianza para la diferencia de proporciones verdadera de estudiantes que les agrade la nueva bebida en ambos Campus.

Parametros resaltantes:
- Muestra 1 ($n_1$) = 500
- Numero de exitos 1 ($X_1$) = 300 
- Muestra 2 ($n_2$) = 400
- Numero de extios 2 ($X_2$) = 230
- Confianza = 95%

**Formulas a utilizar:**
$$
IC= {p_1-p_2-Z_{(1-\frac{\alpha}{2})}\sqrt{\frac{p_1q_1}{n_1}+\frac{p_2q_2}{n_2}}<\pi_1-\pi_2<p_1-p_2-Z_{(1-\frac{\alpha}{2})}\sqrt{\frac{p_1q_1}{n_1}+\frac{p_2q_2}{n_2}}}
$$

A favor:
$$
p=\frac{X}{n}
$$
En contra:
$$
q = 1 - p 
$$

In [27]:
import scipy.stats as st
import math

# Asignando valores:
n1=500
n2=400
x1=300
x2=230
CF=0.95

p1=x1/n1
p2=x2/n2
q1=1-p1
q2=1-p2
a = 1 - CF

# Calculamos Z
Z = round(st.norm.ppf(1-(a/2)),4)

# Aplicamos la formula IC
IC1 = round((p1-p2)-Z*(math.sqrt(p1*q1/n1+p2*q2/n2)),3)
IC2 = round((p1-p2)+Z*(math.sqrt(p1*q1/n1+p2*q2/n2)),3)

print("Z:", Z)
print("Intervalo de confianza:", IC1, "<= pi <=", IC2)

Z: 1.96
Intervalo de confianza: -0.04 <= pi <= 0.09


Con un 95% de confianza la verdadera diferencia de proporciones está entre -0.04 y 0,09.

---

# Intervalos de confianza para la varianza

Los siguientes datos representan las edades que tenían al momento de morir por enfermedad de una muestra de 20 personas de un pueblo:

![datosProblema](../Images/problemaIntervalosCOnfianzaVarianza.png)

Hallar un intervalo de confianza del 95 % para la varianza poblacional de la edad de muerte.


Parametros resaltantes:
- Muestra 1 ($n$) = 20
- Confianza = 95%

**Formula a utilizar:**
Intervalo de confianza para la varianza:
$$
\frac{(n-1)S^2}{X^2_{(1-\frac\alpha2,n-1)}}<\sigma^2<\frac{(n-1)S^2}{X^2_{(\frac\alpha2,n-1)}}
$$

In [36]:
import scipy.stats as st
import numpy as np
import math

# Asignando valores:
n=20
CF=0.95
datos=[80,90,85,82,75,58,70,84,87,81,87,61,73,84,85,70,78,95,77,52]
S= np.var(datos,ddof=1)
a= 1 - CF

# Calculamos X usando la tabla Chi-Cuadrado
X1 = round(st.chi2.ppf(1-(a/2),n-1),4)
X2 = round(st.chi2.ppf(a/2,n-1),4)

# Aplicamos la formula IC
IC1 = round(((n-1)*S/X1),3)
IC2 = round(((n-1)*S/X2),3)

print("X1:", X1)
print("X2:", X2)
print("S:",S)
print("Intervalo de confianza:", IC1, "<= o^2 <=", IC2)

X1: 32.8523
X2: 8.9065
S: 122.1157894736842
Intervalo de confianza: 70.625 <= o^2 <= 260.506


Con un nivel de confianza del 95% podemos afirmar que la varianza de las edades de las personas que mueren por enfermedad en el pueblo se encuentra entre 70.62 y 260.51

---